distributed_training/validator/forward.py

```diff
@@ -73,252 +73,269 @@ async def forward(self):
         all_reduce = False
         self.event.update({"synapse_type": "train"})
 
-    # Get active miners
-    self.miner_uids = await get_random_uids(
-        self,
-        dendrite=self.dendrite,
-        k=sample_size,
-        epoch=self.local_progress.epoch if all_reduce else None,
-    )
-
-    self.event.update({"uids": self.miner_uids})
-    bt.logging.info(f"UIDs:  {self.miner_uids}")
+    if (self.uid == self.master_uid) or (all_reduce == False):
```

Comment:
When `all_reduce == True` only the master validator enters this loop, which is what we're already doing atm without building it in the code. When `all_reduce == False` then all validators enter this loop as send the Train synapse to miners.